codec.go

```diff
@@ -222,13 +236,29 @@ func (d Decimal) MarshalJSON() ([]byte, error) {
 	return []byte(`"` + d.stringBigInt(true) + `"`), nil
 }
 
+// UnmarshalJSON implements the [json.Unmarshaler] interface.
 func (d *Decimal) UnmarshalJSON(data []byte) error {
+	// Remove quotes if they exist.
+	if len(data) > 2 && data[0] == '"' && data[len(data)-1] == '"' {
+		data = data[1 : len(data)-1]
+	}
+
+	return d.UnmarshalText(data)
+}
+
+// MarshalText implements the [encoding.TextMarshaler] interface.
+func (d Decimal) MarshalText() ([]byte, error) {
+	return []byte(d.String()), nil
```

Comment:
I think this method should follow `MarshalJSON` implementation without quote to avoid redundant conversion from byte to string in `d.String()` and back to byte again. 
```
if !d.coef.overflow() {
        // set withQuote to false
	return d.bytesU128(true, false), nil
}

return []byte(d.stringBigInt(true)), nil

```

Reply:
Yes you're totally right, I'll update it now.

Reply:
also can you add a separate test for `UnmarshalText` only, similar to `TestUnmarshalNumber`? The reason is `UnmarshalText` usually accepts external data which can contain unexpected data. The test you added uses data from `MarshalText` and the format is correct already

Reply:
@quagmt I did not add the same test cases for `UnmarshalNumber` as it seems that a fix is required for `UnmarshalJSON` with an empty string.

Would like me to fix it, or should it be in a separate pull request.

Reply:
@quagmt I hope you don't mind, I've added a fix for unmarshalling empty JSON strings and added the other test cases and renamed the test to `TestUnmarshalJSON` as it's now more appropriate.